src/kmer.jl

```diff
@@ -0,0 +1,31 @@
+# kmer.jl
+# =======
+#
+# Basic kmer types.
+#
+# This file is a part of the Kmers.jl, a package in the BioJulia ecosystem.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+abstract AbstractKmer
+
+for inttype in subtypes(Unsigned)
+    nbits = sizeof(inttype) * 8
+    typename = "Kmer$nbits"
+    @eval begin
+        bitstype $nbits $(parse(typename)){T<:NucleicAcid, K} <: AbstractKmer
```

Comment:
I think that should be up to @kdmurray91 as the dedicated maintainer. But that is certainly something that can be done. Note this is still WIP and as you can probably tell from the file, I was trying out your multiple primitive type idea, and considering the pro's and cons vs. a single type parametric for the data field.